TypeSystem/flow-analysis/reachability_A01_t23.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion
+/// - Variable or getter: If `N` is an expression of the form `x` where the
+///   type of `x` is `T` then:
+/// If `T <: Never` then:
+///   - Let `after(N) = unreachable(before(N))`.
+/// Otherwise:
+///   - Let `after(N) = before(N)`.
+///
+/// @description Checks that the code is unreachable after a variable of type
+/// `Never`. Test a `for(;;)` statement in a list literal.
+/// @author sgrekhov22@gmail.com
+/// @issue 60089
+
+void test1(Never n) {
+  late int i;
+  bool b = (() => true)();
+  if (b) {
+    [
+      for (n;;)        // The code after this point is unreachable
+        i = 42,  // Variable is initialized in a dead code. This leaves it definitely unassigned
+      i = 0
+    ];
+  }
+  i; // It is an error to read a local late variable when it is definitely unassigned.
+//^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+void test2(Never n) {
+  late int i;
+  bool b = (() => true)();
+  if (b) {
+    [
+      for (;n;)  // The code after this point is unreachable
+        i = 42,
+      i = 0
+    ];
+  }
+  i; // It is an error to read a local late variable when it is definitely unassigned.
+//^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+void test3(Never n) {
+  late int i;
+  bool b = (() => true)();
+  if (b) {
+    [
+      for (;; n)  // The code after this for statement is unreachable
```

Comment:
Interesting&mdash;I thought the flow analysis wouldn't catch this! It needs to recognize that the condition is implicitly turned into `true`; then it needs to notice that the expression list which is the third element is _guaranteed_ to be executed, and then it will conclude that this for statement will definitely not execute the body.

I hadn't expected the flow analysis to have a notion of "will _definitely_ evaluate to `true`". Anyway, it could be a special treatment of the case where the condition is empty.

However, if this test fails then it seems reasonable to double check the spec and, most likely, adjust the expectation.

Reply:
The test doesn't fail, it works in both the analyzer and the CFE

Reply:
OK, I can see that the `true` literal does indeed get this kind of treatment. Thanks!